conans/model/conf.py

```diff
@@ -323,6 +322,40 @@ def filter_user_modules(self):
                 result._values[k] = v
         return result
 
+    def copy_conaninfo_conf(self):
+        """
+        Get a new `Conf()` object with all the configurations required by the consumer
+        to be included into the final `ConanInfo().package_id()` computation. For instance, let's
+        suppose that we have this Conan `profile`:
+
+        ```
+        ...
+        [conf]
+        tools.info.package_id:confs=["tools.build:cxxflags", "tools.build:cflags"]
+        tools.build:cxxflags=["flag1xx"]
+        tools.build:cflags=["flag1"]
+        tools.build:defines=["DEF1"]
+        ...
+
+        Then, the resulting `Conf()` will have only these configuration lines:
+
+        >>> print(conf.dumps())
+        tools.build:cxxflags=["flag1xx"]
+        tools.build:cflags=["flag1"]
+        ```
+
+        :return: a new `< Conf object >` with the configuration selected by `tools.info.package_id:confs`.
+        """
+        result = Conf()
+        # Reading the list of all the configurations selected by the user to use for the package_id
+        package_id_confs = self.get("tools.info.package_id:confs", default=[], check_type=list)
+        for conf_name in package_id_confs:
+            value = self.get(conf_name)
+            # Pruning None values, those should not affect package ID
+            if value is not None:
+                result.define(conf_name, value)
```

Comment:
Uhm, I don't love it, because semantically, not adding any flag should result in the same ``package_id``. Probably check ``if value:`` for discarding empty strings and things too.

Reply:
oh, yes, you're completely right.